lib/board.js

```diff
@@ -1,37 +1,149 @@
-var blockCoordinates = [{x: 12, y: 12},
-                        {x: 124, y: 12},
-                        {x: 236, y: 12},
-                        {x: 348, y: 12},
-                        {x: 12,  y: 124},
-                        {x: 124, y: 124},
-                        {x: 236, y: 124},
-                        {x: 348, y: 124},
-                        {x: 12,  y: 236},
-                        {x: 124, y: 236},
-                        {x: 236, y: 236},
-                        {x: 348, y: 236},
-                        {x: 12,  y: 348},
-                        {x: 124, y: 348},
-                        {x: 236, y: 348},
-                        {x: 348, y: 348}]
+const StaticNode = require('./static-node');
+const NodeList = require('./node-list');
+
+var blockCoordinates = [{x: 12,  y: 12,  id: 1},
+                        {x: 124, y: 12,  id: 2},
+                        {x: 236, y: 12,  id: 3},
+                        {x: 348, y: 12,  id: 4},
+                        {x: 12,  y: 124, id: 5},
+                        {x: 124, y: 124, id: 6},
+                        {x: 236, y: 124, id: 7},
+                        {x: 348, y: 124, id: 8},
+                        {x: 12,  y: 236, id: 9},
+                        {x: 124, y: 236, id: 10},
+                        {x: 236, y: 236, id: 11},
+                        {x: 348, y: 236, id: 12},
+                        {x: 12,  y: 348, id: 13},
+                        {x: 124, y: 348, id: 14},
+                        {x: 236, y: 348, id: 15},
+                        {x: 348, y: 348, id: 16}];
+
+var nodes = [];
+var rightLists = [];
+var leftLists = [];
+var upLists = [];
+var downLists = [];
+
 
 function setupBoard() {
-  var canvas = document.getElementById("grid")
+  var canvas = document.getElementById("grid");
   var context = canvas.getContext('2d');
+  context.fillStyle="#FFB266";
   // draw a grid
   drawGrid(context);
   // create nodes
+  setupNodeGrid();
   // create list
+  createLists();
   // randomly place two tiles
   placeStartingTiles(context);
 }
 
 function drawGrid(context) {
   blockCoordinates.forEach(function(coordinate) {
     context.strokeRect(coordinate.x, coordinate.y, 100, 100);
-  })
+  });
+}
+
+function setupNodeGrid(){
+  blockCoordinates.forEach(function(coordinate){
+    var newNode = new StaticNode({id: coordinate.id});
+    nodes.push(newNode);
```

Comment:
what is "nodes"? I am just unsure where that came from and would love more context.